### PR TITLE
Fixing issues with encoders/decoders

### PR DESCRIPTION
**Pull request by deusaquilus**
Some stuff to get encoders/decoders working.

### CODE REVIEW

File: quill-jdbc/src/test/scala/io/getquill/context/jdbc/postgres/JdbcEncodingSpec.scala [hunk 1]
```diff
@@ -10,6 +10,9 @@ class JdbcEncodingSpec extends EncodingSpec {
   import testContext._
 
   "encodes and decodes types" in {
+
+    val e = implicitly[Encoder[io.getquill.context.sql.Number]]
```
Comment (deusaquilus) [inline quill-jdbc/src/test/scala/io/getquill/context/jdbc/postgres/JdbcEncodingSpec.scala]: This was just for my own testing. Ignore

File: quill-ndbc-postgres/src/main/scala/io/getquill/context/ndbc/PostgresDecoders.scala [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
 import io.getquill.context.sql.encoding.ArrayEncoding
-import io.getquill.dsl.LowPriorityPostgresImplicits
 import io.trane.ndbc.PostgresRow
 import io.trane.ndbc.value.Value
 
-trait PostgresDecoders extends LowPriorityPostgresImplicits with io.getquill.dsl.LowPriorityImplicits {
+import scala.language.implicitConversions
+import scala.math.BigDecimal.javaBigDecimal2bigDecimal
+
+trait PostgresDecoders {
```
Comment (deusaquilus) [inline quill-ndbc-postgres/src/main/scala/io/getquill/context/ndbc/PostgresDecoders.scala]: Looks like LowPriorityPostgresImplicits and LowPriorityImplicits extensions only need to be there for the encoders. I'm not sure why.

File: quill-ndbc-postgres/src/main/scala/io/getquill/context/ndbc/PostgresEncoders.scala [hunk 3]
```diff
@@ -26,6 +33,8 @@ trait PostgresEncoders extends LowPriorityPostgresImplicits with io.getquill.dsl
       if (v == null) ps.setNull(idx)
       else f(ps)(idx, v.map(ev).toArray[U])
 
+  implicit override def anyValMappedEncoder[I <: AnyVal, O](implicit mapped: MappedEncoding[I, O], encoder: Encoder[O]): Encoder[I] = mappedEncoder
```
Comment (deusaquilus) [inline quill-ndbc-postgres/src/main/scala/io/getquill/context/ndbc/PostgresEncoders.scala]: The trick here is that `anyValMappedEncoder` is in the primary class `PostgresEncoders` which is the 2st place the implicits get looked up and the `mappedEncoder` is an attached trait which gets looked up later. Also, `anyValEncoder` is part of `LowPriorityImplicits`. By attaching `LowPriorityImplicits` to `LowPriorityPostgresImplicits` I made `anyValEncoder` even later in the implicit search order then `mappedEncoder` or `anyValMappedEncoder`. At least that's how I think it works...

Comment (juliano) [reply quill-ndbc-postgres/src/main/scala/io/getquill/context/ndbc/PostgresEncoders.scala]: Thank you very much for your help here :)

**Pull request merged**